src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.Large.Reflection.cs

```diff
@@ -19,13 +19,9 @@ public LargeObjectWithParameterizedConstructorConverterWithReflection()
         {
         }
 
-        internal override bool RequiresDynamicMemberAccessors => true;
-
-        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
-            Justification = "The ctor is marked RequiresUnreferencedCode.")]
-        internal override void Initialize(JsonSerializerOptions options, JsonTypeInfo? jsonTypeInfo = null)
+        [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
```

Comment:
I feel the suppression for RequiresUnreferencedCode was correct before, won't this require adding RUC in many places now?

Reply:
It's only being called by the reflection-based constructor for `JsonTypeInfo`, which is already marked RUC. I think this is better since we avoid accidentally calling this method in sourcegen paths.